historical_data_api_client/cli.py

```diff
@@ -0,0 +1,110 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import json
+import logging
+import os
+from datetime import datetime
+from logging import getLogger
+
+import click
+from click.types import Choice
+from click_default_group import DefaultGroup
+from colorlog.colorlog import ColoredFormatter
+
+from d3a_api_client.utils import get_simulation_id_from_config_uuid, get_auth_headers_for_requests
+from historical_data_api_client.get_historical_data import get_historical_device_profiles
+from historical_data_api_client.utils import validate_correct_datetime_format, DATE_TIME_FORMAT
+
+log = getLogger(__name__)
+
+
+@click.group(name='d3a-api-client', cls=DefaultGroup, default='run', default_if_no_args=True,
+             context_settings={'max_content_width': 120})
+@click.option('-l', '--log-level', type=Choice(list(logging._nameToLevel.keys())), default='INFO',
+              show_default=True, help="Log level")
+def main(log_level):
+    handler = logging.StreamHandler()
+    handler.setLevel(log_level)
+    handler.setFormatter(
+        ColoredFormatter(
+            "%(log_color)s%(asctime)s.%(msecs)03d %(levelname)-8s (%(lineno)4d) %(name)-30s: "
+            "%(message)s%(reset)s",
+            datefmt="%H:%M:%S"
+        )
+    )
+    root_logger = logging.getLogger()
+    root_logger.setLevel(log_level)
+    root_logger.addHandler(handler)
+
+
+@main.command()
+@click.option('-u', '--username', type=str, help="D3A username", required=None)
+@click.option('-p', '--password', type=str, help="D3A password", required=None)
+@click.option('-c', '--configuration-id', type=str, help="configuration uuid", required=True)
```

Comment:
I'm a bit in doubt about this: If I'm not wrong, we're using different naming for the "same" object (the UUID of a configuration). Here we use `configuration-id`, while on the other `cli.py` we use:
```python
@click.option('-s', '--simulation-id', ...)
```
I think it would be nice to pick one, even though I understand that the concept of simulation is more specific than the one of configuration. I personally think that `configuration-uuid` is a bit more pertinent, since the d3a-api-client can be used to connect to collaborations, not simulations.

Reply:
I am a bit confused by your justification and just to be 100 sure, I like to explain again what the difference between simulation_id and configuration_uuid are (you probably knot this, so sorry if this is too verbose):
**configuration_uuid**: Is the uuid of the configuration of any simulation/collaboration/CN
**simulation_id**: (aka jobId) Is the uuid of the simulation/collaboration/CN run and is used for selecting the results of the wanted run of the d3a simulation. Earlier we allowed multiple runs of each simulation and that is why we have those two and not only one uuid. (it will be a major change to get rid of one of them).

Coming to your suggestion: I agree, that we should use the same nomenclature for bot api-clients although the user has to really provide a configuration_uuid here. I wanted to make it easy for them and let them copy/paste the uuid from the URL of the UI. But you are right, I will rename the user-facing parameter and use config_uuid inside of the code. Thanks!